File: tests/benchmark/methods.py
import math

class Point:
  def __init__(self,x,y):
    self.x=x
    self.y=y

  def move(self):
    self.x+=self.dx
    self.y+=self.dy
  def isNear(self,p):
    vx=self.x-p.x
    vx*=vx
    vy=self.y-p.y
    vy*=vy
    return vx+vy<5
  def calcDir(self,p):
    self.dx=p.x-self.x
    self.dy=p.y-self.y
    l=math.sqrt(self.dx*self.dx+self.dy*self.dy)
    self.dx/=l
    self.dy/=l

  def travel(self,dst):
    self.calcDir(dst)
    while not self.isNear(dst):
      self.move()

def f():
  x=0.0
  for i in range(10000):
    src=Point(0.0,0.0)
    dst=Point(100.0,100.0)
    src.travel(dst)
    x+=src.x+src.y
  print(x)
f()

File: deps/zorro/tests/benchmark/methods.lua
require 'class'

Point=class()

function Point:init(x,y)
  self.x=x
  self.y=y
end

function Point:isNear(p)
  local vx=self.x-p.x
  local vy=self.y-p.y
  vx=vx*vx
  vy=vy*vy
  return vx+vy<5
end

function Point:calcDir(p)
  self.dx=p.x-self.x
  self.dy=p.y-self.y
  local l=math.sqrt(self.dx*self.dx+self.dy*self.dy)
  self.dx=self.dx/l
  self.dy=self.dy/l
end

function Point:move()
  self.x=self.x+self.dx
  self.y=self.y+self.dy
end

function Point:travel(dst)
  self:calcDir(dst)
  while not self:isNear(dst) do
    self:move()
  end
end

local x=0.0
for i=1,10000 do
  local src=Point(0.0,0.0)
  local dst=Point(100.0,100.0)
  src:travel(dst)
  x=x+src.x+src.y
end
print(x)
